homeassistant/components/jlrincontrol/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""Support for Jaguar/Land Rover InControl services."""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, \
+    CONF_NAME, CONF_SCAN_INTERVAL
+from homeassistant.helpers.dispatcher import (
+    dispatcher_send
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+
+__LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'jlrincontrol'
+SIGNAL_VEHICLE_SEEN = '{}.vehicle_seen'.format(DOMAIN)
+DATA_KEY = DOMAIN
+CONF_MUTABLE = 'mutable'
+
+MIN_UPDATE_INTERVAL = timedelta(minutes=1)
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=1)
+
+RESOURCES = {
+    'FUEL_LEVEL_PERC': ('sensor', 'Fuel level', 'mdi:fuel', '%'),
+    'DISTANCE_TO_EMPTY_FUEL': ('sensor', 'Range', 'mdi:road', 'km'),
+    'EXT_KILOMETERS_TO_SERVICE': ('sensor', 'Distance to next service',
+                                  'mdi:garage', 'km'),
+    'ODOMETER_METER': ('sensor', 'Odometer', 'mdi:car', 'km'),
+    'DOOR_IS_ALL_DOORS_LOCKED': ('binary_sensor', 'All Doors Locked',
+                                 'mdi:lock', 'lock')
+
+
+}
+
+SIGNAL_STATE_UPDATED = '{}.updated'.format(DOMAIN)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_UPDATE_INTERVAL):
+            vol.All(cv.time_period, vol.Clamp(min=MIN_UPDATE_INTERVAL)),
+        vol.Required(CONF_NAME, default={}): vol.Schema(
+            {cv.slug: cv.string}),
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the jlrpy component."""
+    import jlrpy
+
+    username = config[DOMAIN].get(CONF_USERNAME)
+    password = config[DOMAIN].get(CONF_PASSWORD)
+
+    state = hass.data[DATA_KEY] = JLRData(config)
+
+    interval = config[DOMAIN].get(CONF_SCAN_INTERVAL)
+
+    connection = jlrpy.Connection(username, password)
```

Comment:
The library should check if credentials are correct. If they are not we should log error and return False here.